deepdrivemd/agents/stream/dbscan.py

```diff
@@ -0,0 +1,698 @@
+import random
+import numpy as np
+import glob
+import subprocess
+import time
+import sys
+import os
+import argparse
+import itertools
+from typing import List, Tuple
+from numba import cuda
+
+from deepdrivemd.utils import Timer, timer, t1Dto2D
+from deepdrivemd.agents.stream.config import OutlierDetectionConfig
+import tensorflow.keras.backend as K
+
+import pickle
+from deepdrivemd.data.stream.OutlierDB import OutlierDB
+from lockfile import LockFile
+from deepdrivemd.data.stream.aggregator_reader import (
+    Streams,
+    StreamVariable,
+    StreamContactMapVariable,
+    StreamScalarVariable,
+)
+
+import cupy as cp
+from cuml import DBSCAN as DBSCAN
+
+from deepdrivemd.models.keras_cvae.model import CVAE
+from simtk.openmm.app.pdbfile import PDBFile
+
+import adios2
+
+
+def clear_gpu():
+    K.clear_session()
+    try:
+        device = int(os.environ["CUDA_VISIBLE_DEVICES"])
+        print("device = ", device)
+        cuda.select_device(device)
+        cuda.close()
+    except Exception as e:
+        print(e)
+
+
+def build_model(cfg: OutlierDetectionConfig, model_path: str):
+    cvae = CVAE(
+        image_size=cfg.final_shape,
+        channels=cfg.final_shape[-1],
```

Comment:
`final_shape` needs to be added to `OutlierDetectionConfig `

Reply:
Done